### PR TITLE
update for sisl change sc-to-lattice

### PR DESCRIPTION
**Pull request by tfrederiksen**
This PR makes change to conform with the `sc-to-lattice` change now in sisl https://github.com/zerothi/sisl/pull/550.

### CODE REVIEW

**Comment by zerothi** [comment]
This might have the problem that you'll need to bump the sisl version, if that is what you want that is fine, but users who are installing sisl (via pip), and then using hubbard will have problems since it is not released yet.

**Comment by zerothi** [comment]
I am making a commit now that should re-enable the `Geometry.sc` handling, it would be nice if you could check that hubbard works, even with the lattice change.

**Comment by zerothi** [comment]
I have tried some of the examples, and they seem to work just fine! :) Great!

They raise a warning which they should! :)

File: hubbard/grid.py [hunk 1]
```diff
@@ -28,11 +28,11 @@ def real_space_grid(geometry, sc, vector, shape, mode='wavefunction', **kwargs):
     # Create a temporary copy of the geometry
     g = geometry.copy()
 
-    # Set new sc to create real-space grid
-    g.set_sc(sc)
+    # Set new lattice to create real-space grid
+    g.set_sc(lattice)
```
Comment (zerothi) [inline hubbard/grid.py:32]: this should be `set_lattice`

Comment (tfrederiksen) [reply hubbard/grid.py]: done, thanks!

**Comment by zerothi** [comment]
There are still many references to SuperCell...

**Comment by tfrederiksen** [comment]
> There are still many references to SuperCell...

OK, let's wait with this PR to after the workshop.

**Comment by zerothi** [comment]
I would advice you to get this in, at some point (not too far) the `sc` arguments will probably die... ;)

**Comment by tfrederiksen** [comment]
Yes, let's merge this Sofia.

**Comment by sofiasanz** [comment]
Dear both, I agree that we should update the `hubbard` package along with `sisl`. I'm sorry but I was a bit behind all the updates from `sisl` and now I am running the test files and apparently there are some issues with `density.calc_n` method. It doesn't prompt an error message but it doesn't obtain the correct numbers. I'm looking into it.

**Comment by zerothi** [comment]
@sofiasanz let me know if you have any questions, sisl has moved quite a bit... To a large extent it *should* be backwards compatible, but I would probably recommend to bump the required version of sisl to something sensible, say 0.15 or something...

**Comment by sofiasanz** [comment]
> @sofiasanz let me know if you have any questions, sisl has moved quite a bit... To a large extent it _should_ be backwards compatible, but I would probably recommend to bump the required version of sisl to something sensible, say 0.15 or something...

Thank you Nick, so far I tried with version 0.15.0 and the error persist, while with version 0.14.3 (the previous release to 0.15.0) the tests work fine, so there must be some change between these two versions that affects `density.calc_n` in a way.

**Comment by sofiasanz** [comment]
In any case, this error is probably not related to this PR, so we can merge this.

**Comment by sofiasanz** [comment]
Dear both, I think that the error came because we didn't update the deprecated argument `sum` for `sisl.physics.electron.norm2` in `sisl >0.15.0`, which now accepts the argument `projection` instead. Since we use this method to obtain the density in `hubbard.density.calc_n` there was an error since it was using the default value for `projection` which was ``"sum"`` apparently. 
I set this value to `projection=' atoms'` and now it seems to work. However, this only works for `sisl>=0.15.0`, so if we implement this change we must ensure that the users are using this version of `sisl`. Any ideas on how to proceed?

**Comment by sofiasanz** [comment]
In any case I will merge this PR sine it doesn't have anything to do with the mentioned error. I will open a new PR and we can discuss it there.

**Pull request merged**